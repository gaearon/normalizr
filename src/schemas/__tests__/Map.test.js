/* eslint-env jest */
// import { fromJS } from 'immutable';
import { denormalize, normalize, schema } from '../../';

describe(`${schema.Map.name} normalization`, () => {
  it('normalizes the values of an object with the given schema', () => {
    const dog = new schema.Entity('dogs');
    const pets = new schema.Map(dog);

    const pet = [
      { id: 1, name: 'fido' },
      { id: 2, type: 'oliver' }
    ];

    expect(normalize(pet, pets)).toMatchSnapshot();
  });
});

describe(`${schema.Map.name} denormalization`, () => {
  it('denormalizes the values of an object with the given schema', () => {
    const dog = new schema.Entity('dogs');
    const pets = new schema.Map(dog);

    const entities = { dogs: {
      1: { id: 1, name: 'fido' },
      2: { id: 2, type: 'oliver' }
    } };

    expect(denormalize({
      '1': 1,
      '2': 2
    }, pets, entities)).toMatchSnapshot();
  });
});
