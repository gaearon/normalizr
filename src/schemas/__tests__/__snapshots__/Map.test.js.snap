// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapSchema denormalization denormalizes the values of an object with the given schema 1`] = `
Array [
  Object {
    "id": 1,
    "name": "fido",
  },
  Object {
    "id": 2,
    "type": "oliver",
  },
]
`;

exports[`MapSchema normalization normalizes the values of an object with the given schema 1`] = `
Object {
  "entities": Object {
    "dogs": Object {
      "1": Object {
        "id": 1,
        "name": "fido",
      },
      "2": Object {
        "id": 2,
        "type": "oliver",
      },
    },
  },
  "result": Object {
    "1": 1,
    "2": 2,
  },
}
`;
